### DISCUSSION TITLE
MPI is not speeding up?

### DISCUSSION THREAD

**Discussion by Zhoujieliang** [NCAR/noahmp#164] [General]
hello,
I compile Noahmp with two version.
One is serial.
![image](https://github.com/user-attachments/assets/e7e781be-cdae-4852-8d04-7e531ada2610)
Another is mpi.
![image](https://github.com/user-attachments/assets/885c439c-0670-4aef-8b30-dc80664f5437)
And use same data, same namelist, there is no difference in speed between the two when running **_./hrldas.exe_**
**_How can I check if I have successfully compiled MPI?_**
![image](https://github.com/user-attachments/assets/f38be1fd-e998-4432-86f9-4b7add556a39)

**Comment by cenlinhe** [comment]
for mpi run, you will need to use the mpi run command instead of just ./hrldas.exe. The mpi run command will be slightly different in different computers or machines (https://www.open-mpi.org/doc/v4.0/man1/mpirun.1.php). you may try "mpirun -np 4 ./hrldas.exe" This means using 4 cores to run the model.

**Reply by Zhoujieliang** [reply]
Thank you, the speed has significantly improved.
Is it normal for the printed information to become four times?
![image](https://github.com/user-attachments/assets/6c234b69-f989-4731-a133-0c40714ff0d3)

**Reply by cenlinhe** [reply]
yes, because you used 4 cores so the info from each core will be displayed. If you use more cores, then more lines will be shown.
